ir/node.cpp

```diff
@@ -94,4 +102,24 @@ Util::JsonObject* IR::Node::sourceInfoJsonObj() const {
     return json;
 }
 
+void IR::Node::sourceInfoToJSON(JSONGenerator &json) const {
+    Util::SourceInfo si = srcInfo;
+    unsigned lineNumber, columnNumber;
+    cstring fName = prepareSourceInfoForJSON(si, &lineNumber, &columnNumber);
+    if (fName == nullptr) {
+        // Same reasoning as above.
+        return;
+    }
+
+    json << "," << std::endl
+         << json.indent++ << "\"Source_Info\" : {" << std::endl;
+
+    json << json.indent << "\"filename\" : " << fName << "," << std::endl;
+    json << json.indent << "\"line\" : " << lineNumber << "," << std::endl;
+    json << json.indent << "\"column\" : " << columnNumber << "," << std::endl;
+    json << json.indent << "\"source_fragment\" : " << si.toBriefSourceFragment() << std::endl;
```

Comment:
Part of the problem is that we're not including the entire source code in the json, so just having start/end positions is not enough; we need the source fragments too.  But with the source fragments, you're right, we probably don't need the end points.  Of course, the source code could readily be packaged separately from the json -- with the start/end positions, a tool could easily extract the fragments from the source code itself.  No good answers.